server/controllers/student.py

```diff
@@ -1,14 +1,118 @@
-from server.models import User, db
+from server.models import User, Course, Assignment, Group, Backup, db
+
 
 student = Blueprint('student', __name__)
 
 
 @student.route("/")
 @login_required
-def index():
-    return "student!", 200
+def index(auto_redir=True):
+    enrollments = current_user.enrollments(VALID_ROLES)
+    student_enrollments = [e for e in enrollments if e.role == STUDENT_ROLE]
+    all_courses = Course.query.all()
+    courses = {
+        'instructor': [e.course for e in enrollments if e.role in STAFF_ROLES],
+        'current': [e.course for e in student_enrollments if e.course.active],
+        'past': [e.course for e in student_enrollments if not e.course.active],
+        'all': all_courses,
+        'num_enrolled': len(enrollments)
+    }
+    # Make the choice for users in one course
+    if len(enrollments) == 1 and auto_redir:
+        return redirect(url_for(".course", cid=enrollments[0].course_id))
+    return render_template('student/courses/index.html', **courses)
+
+
+def is_enrolled(func):
+    """ A decorator for routes to ensure that user is enrolled in
+    the course. Gets the course id from the named arg cid of the route.
+    A user is enrolled if they are participating in the course with any role.
+
+    Usage:
+    @is_enrolled # Get the course id from the cid param of the routes
+    def my_route(cid): ...
+    """
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        course_id = kwargs['cid']
+        enrolled = current_user.is_enrolled(course_id)
+        if not enrolled and not current_user.is_admin:
+            flash("You have not been added to this course on OK", "warning")
+        return func(*args, **kwargs)
+    return wrapper
+
+
+@student.route("/course/<int:cid>")
+@login_required
+@is_enrolled
+def course(cid):
+    course = Course.query.get(cid)
+    # TODO : Should consider group submissions as well.
+    user_id = current_user.id
+    assignments = {
+        'active': [(a, a.submission_time(user_id), a.group(user_id)) \
+                        for a in course.assignments if a.active],
+        'inactive': [(a, a.submission_time(user_id), a.group(user_id)) \
+                            for a in course.assignments if not a.active]
+    }
+    return render_template('student/course/index.html', course=course,
+                           **assignments)
+
+@student.route("/course/<int:cid>/assignment/<int:aid>")
+@login_required
+@is_enrolled
+def assignment(cid, aid):
+    assgn = Assignment.query.filter_by(id=aid, course_id=cid).one_or_none()
+    if assgn:
+        course = assgn.course
+        group = Group.lookup(current_user, assgn)
+        if group:
+            # usr_ids = [u.id for u in group.users()]
+            # TODO : Fetch backups from group.
+            pass
+        backups = assgn.backups(current_user.id).limit(5).all()
+        subms = assgn.submissions(current_user.id).limit(5).all()
+        flagged = any([s.flagged for s in subms])
```

Comment:
Just saw it. That looks good 
Relevant line: https://github.com/Cal-CS-61A-Staff/ok/commit/21c1764f82fc7063cce7f0929dc86d8f19c7b95d#diff-e07bdcf29922d0f33a2c70e8145df054R188